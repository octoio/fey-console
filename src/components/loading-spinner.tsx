import { Spin } from "antd";
import React from "react";

interface LoadingSpinnerProps {
  size?: "small" | "default" | "large";
  tip?: string;
  height?: string;
}

export const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
  size = "large",
  tip = "Loading...",
  height = "400px",
}) => (
  <div
    data-testid="loading-spinner"
    style={{
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      height,
    }}
  >
    <Spin size={size} tip={tip} />
  </div>
);
