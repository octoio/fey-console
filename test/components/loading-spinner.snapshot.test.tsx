import { describe, it, expect } from "vitest";
import { LoadingSpinner } from "@components/loading-spinner";
import { render } from "@testing-library/react";

describe("LoadingSpinner Snapshots", () => {
  describe("Default Props Snapshots", () => {
    it("should match snapshot with default props", () => {
      const { container } = render(<LoadingSpinner />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Size Variations Snapshots", () => {
    it("should match snapshot with small size", () => {
      const { container } = render(<LoadingSpinner size="small" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with default size", () => {
      const { container } = render(<LoadingSpinner size="default" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with large size", () => {
      const { container } = render(<LoadingSpinner size="large" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Tip Text Variations Snapshots", () => {
    it("should match snapshot with custom tip", () => {
      const { container } = render(<LoadingSpinner tip="Please wait..." />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with empty tip", () => {
      const { container } = render(<LoadingSpinner tip="" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with special characters in tip", () => {
      const { container } = render(
        <LoadingSpinner tip="Loading... 🔄 (50%)" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with long tip text", () => {
      const { container } = render(
        <LoadingSpinner tip="This is a very long loading message that might wrap to multiple lines" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Height Variations Snapshots", () => {
    it("should match snapshot with custom pixel height", () => {
      const { container } = render(<LoadingSpinner height="600px" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with percentage height", () => {
      const { container } = render(<LoadingSpinner height="50%" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with viewport height", () => {
      const { container } = render(<LoadingSpinner height="100vh" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with calc height", () => {
      const { container } = render(
        <LoadingSpinner height="calc(100% - 50px)" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with small height", () => {
      const { container } = render(<LoadingSpinner height="100px" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Combined Props Snapshots", () => {
    it("should match snapshot with small size and custom tip", () => {
      const { container } = render(
        <LoadingSpinner size="small" tip="Saving..." />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with all custom props", () => {
      const { container } = render(
        <LoadingSpinner
          size="default"
          tip="Processing data..."
          height="300px"
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with large size, long tip, and custom height", () => {
      const { container } = render(
        <LoadingSpinner
          size="large"
          tip="Loading large dataset, this may take a few moments..."
          height="500px"
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with minimal height and no tip", () => {
      const { container } = render(
        <LoadingSpinner size="small" tip="" height="50px" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Edge Cases Snapshots", () => {
    it("should match snapshot with zero height", () => {
      const { container } = render(<LoadingSpinner height="0px" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with very large height", () => {
      const { container } = render(<LoadingSpinner height="9999px" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with numeric characters in tip", () => {
      const { container } = render(<LoadingSpinner tip="Loading 12345..." />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with special unicode characters", () => {
      const { container } = render(
        <LoadingSpinner tip="Загрузка... ☕ 読み込み中" />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Multiple Instances Snapshots", () => {
    it("should match snapshot with multiple different spinners", () => {
      const { container } = render(
        <div>
          <LoadingSpinner size="small" tip="Loading 1" height="200px" />
          <LoadingSpinner size="default" tip="Loading 2" height="300px" />
          <LoadingSpinner size="large" tip="Loading 3" height="400px" />
        </div>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with identical spinners", () => {
      const { container } = render(
        <div>
          <LoadingSpinner />
          <LoadingSpinner />
        </div>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("Component Structure Snapshots", () => {
    it("should match snapshot showing complete component structure", () => {
      const { container } = render(
        <LoadingSpinner
          size="large"
          tip="Complete structure test"
          height="400px"
        />,
      );
      // Capture the entire component tree
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot of wrapper div only", () => {
      const { container } = render(<LoadingSpinner />);
      // Focus on just the wrapper div
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
