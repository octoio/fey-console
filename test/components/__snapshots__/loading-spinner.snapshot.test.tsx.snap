// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner Snapshots > Combined Props Snapshots > should match snapshot with all custom props 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 300px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Combined Props Snapshots > should match snapshot with large size, long tip, and custom height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 500px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Combined Props Snapshots > should match snapshot with minimal height and no tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 50px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-sm ant-spin-spinning css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Combined Props Snapshots > should match snapshot with small size and custom tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-sm ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Component Structure Snapshots > should match snapshot of wrapper div only 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Component Structure Snapshots > should match snapshot showing complete component structure 1`] = `
<div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 400px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Default Props Snapshots > should match snapshot with default props 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Edge Cases Snapshots > should match snapshot with numeric characters in tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Edge Cases Snapshots > should match snapshot with special unicode characters 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Edge Cases Snapshots > should match snapshot with very large height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 9999px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Edge Cases Snapshots > should match snapshot with zero height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 0px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Height Variations Snapshots > should match snapshot with calc height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: calc(100% - 50px);"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Height Variations Snapshots > should match snapshot with custom pixel height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 600px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Height Variations Snapshots > should match snapshot with percentage height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 50%;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Height Variations Snapshots > should match snapshot with small height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 100px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Height Variations Snapshots > should match snapshot with viewport height 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 100vh;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Multiple Instances Snapshots > should match snapshot with identical spinners 1`] = `
<div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 400px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 400px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Multiple Instances Snapshots > should match snapshot with multiple different spinners 1`] = `
<div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 200px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-sm ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 300px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
  <div
    data-testid="loading-spinner"
    style="display: flex; justify-content: center; align-items: center; height: 400px;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
    >
      <span
        class="ant-spin-dot-holder"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Size Variations Snapshots > should match snapshot with default size 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Size Variations Snapshots > should match snapshot with large size 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Size Variations Snapshots > should match snapshot with small size 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-sm ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Tip Text Variations Snapshots > should match snapshot with custom tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Tip Text Variations Snapshots > should match snapshot with empty tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Tip Text Variations Snapshots > should match snapshot with long tip text 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner Snapshots > Tip Text Variations Snapshots > should match snapshot with special characters in tip 1`] = `
<div
  data-testid="loading-spinner"
  style="display: flex; justify-content: center; align-items: center; height: 400px;"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-tjsggz"
  >
    <span
      class="ant-spin-dot-holder"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </span>
  </div>
</div>
`;
